libcontainer/rootfs_linux.go

```diff
@@ -152,12 +153,24 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		}
 		return nil
 	case "tmpfs":
+		copyUp := m.Extensions&configs.EXT_COPYUP == configs.EXT_COPYUP
+		tmpDir := ""
 		stat, err := os.Stat(dest)
 		if err != nil {
 			if err := os.MkdirAll(dest, 0755); err != nil {
 				return err
 			}
 		}
+		if copyUp {
+			tmpDir, err = ioutil.TempDir("/run", "runctmpdir")
```

Comment:
Why don't we do it like this:

1. Create a tmpfs somewhere (`/tmp/runc.XXXXX`).
2. Copy the contents of the directory we're going to dump the `tmpfs` on top of.
3. Bind-mount it over the directory.
4. Lazy unmount and remove the initial temporary directory.

That way, we don't have to do any double-copying.